ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/PartyManagementServiceIT.scala

```diff
@@ -163,48 +162,16 @@ class PartyManagementServiceIT
 
       "create unique party names when allocating many parties" in allFixtures { c =>
         val client = partyManagementClient(c.partyManagementService)
+        def allocateParty(i: Int) = client.allocateParty(None, Some(s"Test party $i"))
         val N = 100
-
         for {
           initialParties <- client.listKnownParties()
-          // Note: The following call concurrently creates N parties
-          resultEs <- Future.traverse(1 to N)(i =>
-            withGrpcError(client.allocateParty(None, Some(s"Test party $i"))))
+          results <- Future.traverse(1 to N)(allocateParty)
           finalParties <- client.listKnownParties()
         } yield {
-
```

Comment:
Party allocation may fail with `UNIMPLEMENTED` or `INVALID_ARGUMENT`, see https://github.com/digital-asset/daml/blob/master/ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/party_management_service.proto#L47. 

This removed code was responsible for checking that the ledger behaves sensibly in case it returns a gRPC error for `allocateParty`. Are we ok with this test only succeeding for ledgers where `allocatedParty` always succeeds? Could we potentially amend the protobuf description, e.g., to specify that the call may not be rejected because of a missing hint, or because of a malformed description?

Reply:
I see. I mostly re-worked the tests within this draft PR because the test was actually failing due to a specific issue I wanted to single out. I would say that we want to check that either all calls return successfully or that all calls return `UNIMPLEMENTED`. It would be probably be best if we could somehow signal the latter situation via a different cue, e.g. a cancelled test.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2040/commits/d487cd9a5775aff2e47dd2219b2f9e3859b97677